src/sources/fluent/mod.rs

```diff
@@ -168,27 +175,55 @@ impl FluentDecoder {
         byte_size: usize,
     ) -> Result<(), DecodeError> {
         match message {
-            FluentMessage::Message(tag, timestamp, record)
-            | FluentMessage::MessageWithOptions(tag, timestamp, record, ..) => {
+            FluentMessage::Message(tag, timestamp, record) => {
                 self.unread_frames.push_back((
                     FluentFrame {
                         tag,
                         timestamp,
                         record,
+                        chunk: None,
                     },
                     byte_size,
                 ));
                 Ok(())
             }
-            FluentMessage::Forward(tag, entries)
-            | FluentMessage::ForwardWithOptions(tag, entries, ..) => {
+            FluentMessage::MessageWithOptions(tag, timestamp, record, options) => {
+                self.unread_frames.push_back((
+                    FluentFrame {
+                        tag,
+                        timestamp,
+                        record,
+                        chunk: options.chunk,
+                    },
+                    byte_size,
+                ));
+                Ok(())
+            }
+            FluentMessage::Forward(tag, entries) => {
                 self.unread_frames.extend(entries.into_iter().map(
                     |FluentEntry(timestamp, record)| {
                         (
                             FluentFrame {
                                 tag: tag.clone(),
                                 timestamp,
                                 record,
+                                chunk: None,
+                            },
+                            byte_size,
+                        )
+                    },
+                ));
+                Ok(())
+            }
+            FluentMessage::ForwardWithOptions(tag, entries, options) => {
+                self.unread_frames.extend(entries.into_iter().map(
+                    |FluentEntry(timestamp, record)| {
+                        (
+                            FluentFrame {
+                                tag: tag.clone(),
+                                timestamp,
+                                record,
+                                chunk: options.chunk.clone(),
```

Comment:
Can we assume ownership of the chunk field here?

Reply:
That is inside a `map` over an iterator, so it may be owned multiple times. So, no.